pkg/nsx/util/utils.go

```diff
@@ -561,6 +561,8 @@ func CasttoPointer(obj interface{}) interface{} {
 		return &v
 	case model.Vpc:
 		return &v
+	case model.LBService:
+		return &v
```

Comment:
Is there any comment to inform developers to change here for each new resource type? @dantingl @lxiaopei

Reply:
If type is missing, operator will crash when populate non-empty cache.

Reply:
We missed `PrincipalIdentity` here